File: HackerNewsTests/HNStoryManagerSpecs.m
//
//  HackerNewsTests.m
//  HackerNewsTests
//
//  Created by xes on 10/15/14.
//  Copyright (c) 2014 wrkstrm. All rights reserved.
//

#import <Quick/Quick.h>
#import <Nimble/Nimble.h>
#import "HNStoryManager.h"

QuickSpecBegin(HNStoryManagerSpecs)
describe(@"HNStoryManager", ^{
    it(@"boots nicely", ^{
        expect(HNStoryManager.new).toNot(beNil());
    });
    
    describe(@"Initializes", ^{
        __block HNStoryManager *manager;
        beforeEach(^{
            manager = HNStoryManager.new;
        });
        
        it(@"CurrentUser", ^{
            expect(manager.currentUser).toNot(beNil());
        });
        
        it(@"UserDatabase", ^{
            HNUser *user = manager.currentUser;
            expect(user.localDatabase).toNot(beNil());
        });
        
        it(@"TopStoriesDocument", ^{
            CBLDatabase *db = manager.currentUser.localDatabase;
            expect(@([db documentCount])).toNot(equal(@0));
        });
        
        it(@"Current Top Stories:", ^{
            expect(manager.currentTopStories).toNot(beNil());
        });
    });
});

QuickSpecEnd

File: HackerNewsTests/HNStoryManagerSpecs.swift
//
//  HackerNewsTests.swift
//  HackerNews
//
//  Created by Cristian Monterroza on 11/14/14.
//  Copyright (c) 2014 wrkstrm. All rights reserved.
//

import Foundation
import XCTest
import Quick
import Nimble

class HNStoryManagerSpecs: QuickSpec {
    override func spec() {
        describe("Initialization.") {
            it ("Boots.") {
                let manager = HNStoryManager()
                expect(true).to(beTrue())
            }
        }
    }
}
